### PR TITLE
Matching image digest between `latest` tag and latest release tag of `rti` image on DockerHub

### PR DESCRIPTION
**Pull request by Depetrol**
Match RTI docker image digest by building only once and push to Docker Hub as two tags when doing release

### CODE REVIEW

File: .github/actions/push-rti-docker/action.yml [hunk 1]
```diff
@@ -11,6 +11,10 @@ inputs:
   DOCKERHUB_TOKEN:
     description: 'The token to log in to Docker Hub'
     required: true
+  latest:
+    description: 'Also push as latest tag or not, set to true of false'
```
Comment (lhstrh) [inline .github/actions/push-rti-docker/action.yml:15]:
```suggestion
    description: 'Also push as latest tag if true'
```

File: .github/actions/push-rti-docker/action.yml [hunk 2]
```diff
@@ -11,6 +11,10 @@ inputs:
...
+    required: true
```
Comment (lhstrh) [inline .github/actions/push-rti-docker/action.yml:16]: Not necessary if a default has been provided...
```suggestion
```

File: .github/actions/push-rti-docker/action.yml [hunk 3]
```diff
@@ -11,6 +11,10 @@ inputs:
...
+    default: 'false'
```
Comment (lhstrh) [inline .github/actions/push-rti-docker/action.yml:17]: I don't think bools need to be quoted?
```suggestion
    default: false
```

File: .github/actions/push-rti-docker/action.yml [hunk 4]
```diff
@@ -27,10 +31,20 @@ runs:
         submodules: recursive
         fetch-depth: 0
     - name: Build and push
+      if: ${{ inputs.latest == 'false' }}
```
Comment (lhstrh) [inline .github/actions/push-rti-docker/action.yml:34]: Just as a matter of style, I prefer to have the `if` at the end of the block.

File: .github/actions/push-rti-docker/action.yml [hunk 5]
```diff
@@ -27,10 +31,20 @@ runs:
...
       uses: docker/build-push-action@v6
       with:
         file: ./core/src/main/resources/lib/c/reactor-c/core/federated/RTI/rti.Dockerfile
         context: ./core/src/main/resources/lib/c/reactor-c
         platforms: linux/amd64, linux/arm64, linux/arm/v7, linux/riscv64
         push: true
         tags: lflang/rti:${{ inputs.tag }}
+    - name: Build and push as latest
+      if: ${{ inputs.latest == 'true' }}
```
Comment (lhstrh) [inline .github/actions/push-rti-docker/action.yml:43]: Just as a matter of style, I prefer to have the `if` at the end of the block.

File: .github/workflows/rti-docker.yml [hunk 6]
```diff
@@ -24,15 +24,19 @@ jobs:
           echo "lf_version=$LF_VERSION"
           echo "lf_version=$LF_VERSION" >> $GITHUB_ENV
       - name: Build and push RTI to Docker Hub
+        if: ${{ endsWith(env.lf_version, '-snapshot') }}
```
Comment (lhstrh) [inline .github/workflows/rti-docker.yml:27]: Just as a matter of style, I prefer to have the `if` at the end of the block.

File: .github/workflows/rti-docker.yml [hunk 7]
```diff
@@ -24,15 +24,19 @@ jobs:
...
         uses: ./.github/actions/push-rti-docker
         with:
           tag: ${{ env.lf_version }}
+          latest: false
           DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
           DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+
       - name: Update latest (released versions only)
+        if: ${{ !endsWith(env.lf_version, '-snapshot') }}
```
Comment (lhstrh) [inline .github/workflows/rti-docker.yml:36]: Just as a matter of style, I prefer to have the `if` at the end of the block.

**Review by lhstrh — APPROVED** [review]
This looks great! It looks like this should work. My comments are only cosmetic, but please apply the requested changes and this should be ready to merge.

File: .github/actions/push-rti-docker/action.yml (hunk 3 above)
Comment (Depetrol) [reply .github/actions/push-rti-docker/action.yml:16]: IntelliJ idea give an error without quotes -- I think this field has to be string

**Comment by lhstrh** [comment]
Perfect, I you could now do an interactive rebase to clean up the history and get rid of superfluous commits, we can merge.

**Pull request merged**